assets/js/modules/analytics/datastore/report.js

```diff
@@ -98,7 +98,7 @@ const baseResolvers = {
 		const { error } = yield fetchGetReportStore.actions.fetchGetReport( options );
 
 		if ( normalizeReportOptions( options ).metrics.some( ( { expression } ) => /^ga:adsense/.test( expression ) ) ) {
-			const isRestrictedMetricError = error?.code === 400 && error?.message.startsWith( 'Restricted metric' );
+			const isRestrictedMetricError = error?.code === 400 && error.message && error.message.startsWith( 'Restricted metric' ) && /ga:adsense/.test( error.message );
```

Comment:
This line is getting a bit long, why un-chain the message?
```suggestion
			const isRestrictedMetricError = error?.code === 400 && error?.message?.startsWith( 'Restricted metric' ) && /ga:adsense/.test( error.message );
```

Reply:
I changed that because of the following clause, which shouldn't run the regular expression test against `undefined`.